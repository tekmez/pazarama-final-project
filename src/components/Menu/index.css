.menu-top-container{
    display: flex;
}
.search-form, .admin-panel, .home{
    height: 100%;
    background: rgba( 255, 255, 255, 255);
    border-radius: 25px;
    display: flex;
    margin-right: 3em;
}
.admin-panel{
    margin-right: 0;
}
.search-btn, .admin-btn, .home-btn{
    border: 0;
    background: 0;
    cursor: pointer;
    transition: 0.1s;
    margin: 0;
    padding: 0;
}
.admin-btn{
    font-size: 18px;
    padding-left: 1em;
}
.search-btn:hover, .admin-btn:hover, .home-btn:hover{
    transform: scale(1.1);
}
.search-icon, .admin-icon, .home-icon{
    margin-right: 1em;
    font-size: 20px;
}
.admin-icon, .home-icon{
    padding-left: 1em;
}
.search-icon{
    margin-left: 1em;
}
