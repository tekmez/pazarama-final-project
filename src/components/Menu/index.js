import React from "react";
import {
  faSearch,
  faUserShield,
  faHome,
} from "@fortawesome/free-solid-svg-icons";
import { useNavigate } from "react-router-dom";
import "./index.css";
import Button from "../Button";

function Menu() {
  const navigate = useNavigate();
  return (
    <div className="menu-top-container">
      <div className="search-form">
        <Button
          icon={faSearch}
          className="search-icon"
          classNameBtn="search-btn"
          onClick={() => navigate("/SearchFormPage")}
        />
      </div>
      <div className="home">
        <Button
          icon={faHome}
          classNameBtn="home-btn"
          className="home-icon"
          onClick={() => navigate("/")}
        />
      </div>
      <div className="admin-panel">
        <Button
          text="Admin"
          icon={faUserShield}
          className="admin-icon"
          classNameBtn="admin-btn"
          onClick={() => navigate("/LoginPage")}
        />
      </div>
    </div>
  );
}

export default Menu;
