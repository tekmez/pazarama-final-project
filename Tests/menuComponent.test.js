import React from "react";
import renderer from "react-test-renderer";
import Menu from "../src/components/Menu/index";
test("Menu Component", () => {
  const tree = renderer
    .create(
      <Menu />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});