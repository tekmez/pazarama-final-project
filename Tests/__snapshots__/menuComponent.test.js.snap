// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component 1`] = `
<div
  className="menu-top-container"
>
  <div
    className="search-form"
  >
    <div
      className="btn-container"
    >
      <button
        className="search-btn"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 search-icon"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="home"
  >
    <div
      className="btn-container"
    >
      <button
        className="home-btn"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-home fa-w-18 home-icon"
          data-icon="home"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="admin-panel"
  >
    <div
      className="btn-container"
    >
      <button
        className="admin-btn"
        onClick={[Function]}
        type="button"
      >
        Admin
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user-shield fa-w-20 admin-icon"
          data-icon="user-shield"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M622.3 271.1l-115.2-45c-4.1-1.6-12.6-3.7-22.2 0l-115.2 45c-10.7 4.2-17.7 14-17.7 24.9 0 111.6 68.7 188.8 132.9 213.9 9.6 3.7 18 1.6 22.2 0C558.4 489.9 640 420.5 640 296c0-10.9-7-20.7-17.7-24.9zM496 462.4V273.3l95.5 37.3c-5.6 87.1-60.9 135.4-95.5 151.8zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm96 40c0-2.5.8-4.8 1.1-7.2-2.5-.1-4.9-.8-7.5-.8h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c6.8 0 13.3-1.5 19.2-4-54-42.9-99.2-116.7-99.2-212z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
